vendor/angle-bracket-invocation-polyfill/runtime-polyfill.js

```diff
@@ -371,36 +371,39 @@ import { lte, gte } from 'ember-compatibility-helpers';
 
               let { manager } = definition;
 
-              let ORIGINAL_DID_CREATE_ELEMENT = manager.didCreateElement;
-              manager.didCreateElement = function(bucket, element, operations) {
-                ORIGINAL_DID_CREATE_ELEMENT.apply(this, arguments);
-                let { args } = bucket;
+              if (!manager.didCreateElement.__IS_ANGLE_BRACKET_PATCHED__) {
+                let ORIGINAL_DID_CREATE_ELEMENT = manager.didCreateElement;
+                manager.didCreateElement = function(bucket, element, operations) {
+                  ORIGINAL_DID_CREATE_ELEMENT.apply(this, arguments);
+                  let { args } = bucket;
 
-                if (lte('2.15.0-beta.1')) {
-                  args = args.namedArgs;
-                }
+                  if (lte('2.15.0-beta.1')) {
+                    args = args.namedArgs;
+                  }
 
-                // on < 2.15 `namedArgs` is only present when there were arguments
-                if (args && args.has('__ANGLE_ATTRS__')) {
-                  let attributeReferences = args.get('__ANGLE_ATTRS__');
-                  let snapshot = attributeReferences.value();
-                  if (snapshot) {
-                    let names = Object.keys(snapshot);
-                    for (let i = 0; i < names.length; i++) {
-                      let attributeName = names[i];
-                      let attributeReference = attributeReferences.get(attributeName);
-
-                      operations.addDynamicAttribute(
-                        element,
-                        attributeName,
-                        attributeReference,
-                        false,
-                        null
-                      );
+                  // on < 2.15 `namedArgs` is only present when there were arguments
+                  if (args && args.has('__ANGLE_ATTRS__')) {
+                    let attributeReferences = args.get('__ANGLE_ATTRS__');
+                    let snapshot = attributeReferences.value();
+                    if (snapshot) {
+                      let names = Object.keys(snapshot);
+                      for (let i = 0; i < names.length; i++) {
+                        let attributeName = names[i];
+                        let attributeReference = attributeReferences.get(attributeName);
+
+                        operations.addDynamicAttribute(
+                          element,
+                          attributeName,
+                          attributeReference,
+                          false,
+                          null
+                        );
+                      }
                     }
                   }
-                }
-              };
+                };
+                manager.didCreateElement.__IS_ANGLE_BRACKET_PATCHED__ = true;
```

Comment:
Hm, I wonder if there is a significant performance concern to setting a property on the function and a `Map` would not mitigate it.

Even the use of a `Map` instead of simply a boolean in module scope implies there could be multiple managers coming through this codepath.

Reply:
Yeah, I would have tried to use a `WeakMap` but I was following the style of the similar fix in https://github.com/ember-polyfills/ember-angle-bracket-invocation-polyfill/pull/55 which was good enough for @rwjblue. We could follow up and change them both if it makes sense to.

If I recall correctly, I think that v8 handles adding one or two properties to a function pretty well. I don't know if that's still true, or true in other engines :shrug: